src/js/actions/transform.js

```diff
@@ -1124,9 +1124,18 @@ define(function (require, exports) {
             this.dispatch(events.ui.TOGGLE_OVERLAYS, { enabled: true });
 
             var appStore = this.flux.store("application"),
-                currentDoc = appStore.getCurrentDocument();
+                currentDoc = appStore.getCurrentDocument(),
+                textLayers = currentDoc.layers.allSelected.filter(function (layer) {
+                    // Reset these layers completely because their impliedFontSize may have changed
+                    return layer.kind === layer.layerKinds.TEXT;
+                }),
+                otherLayers = currentDoc.layers.allSelected.filterNot(function (layer) {
+                    return layer.kind === layer.layerKinds.TEXT;
+                }),
+                textLayersPromise = this.flux.actions.layers.resetLayers(currentDoc, textLayers),
+                otherLayersPromise = this.flux.actions.layers.resetBounds(currentDoc, otherLayers);
```

Comment:
I guess the right thing will happen downstream if there are zero "non-text selected layers" but it still catches my attention that we may be calling this action needlessly in many cases.